src/__test__/calculate.test.js

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { render, fireEvent, screen } from '@testing-library/react';
+import Calculator from '../components/Calculator';
+
+test('renders calculator', () => {
+  render(<Calculator />);
+  const displays = screen.getAllByText('0');
+  expect(displays.length).toBeGreaterThanOrEqual(1);
+});
+
+test('clicking AC button resets the display to "0"', () => {
+  render(<Calculator />);
+  fireEvent.click(screen.getByText('AC'));
+  // Check if at least one display resets to "0"
+  const displays = screen.getAllByText('0');
+  expect(displays.length).toBeGreaterThanOrEqual(1);
+});
+
+test('clicking number buttons updates the display', () => {
+  render(<Calculator />);
+  // Click on the number buttons '1' and '2'
+  fireEvent.click(screen.getByText('1'));
+  fireEvent.click(screen.getByText('2'));
+  // Check if at least one display shows '12'
+  const displays = screen.getAllByText('12');
+  expect(displays.length).toBeGreaterThanOrEqual(1);
+});
```

Comment:
- [ ] Once more let's kindly make sure that all the functionalities and scenarios of calculate.js module should be covered using jest. For this specific, you'll kindly make sure the following scenarios are covered

- 1

```
if (isNumber(buttonName)) {
    if (obj.operation === null) {
      if (obj.next === null) {
        return { ...obj, next: buttonName };
      } if (obj.next === '0') {
        return { ...obj, next: buttonName };
      }
      return { ...obj, next: obj.next + buttonName };
    }
    if (obj.next === null) {
      return { ...obj, next: buttonName };
    }
    return { ...obj, next: obj.next + buttonName };
  }
```

- 2

```
  if (buttonName === '.') {
    if (obj.next === null) {
      return { ...obj, next: '0.' };
    }
    if (!obj.next.includes('.')) {
      return { ...obj, next: `${obj.next}.` };
    }
    return { ...obj };
  }
```

- 3

```
if (buttonName === '=') {
    if (obj.next !== null && obj.operation !== null) {
      try {
        const total = operate(obj.total, obj.next, obj.operation);
        return {
          total: total.toString(),
          next: null,
          operation: null,
        };
      } catch (error) {
        return {
          total: 'Error',
          next: null,
          operation: null,
        };
      }
    }
    return { ...obj };
  }
```

- 4

```
  if (buttonName === '+/-') {
    if (obj.next !== null) {
      return { ...obj, next: (-parseFloat(obj.next)).toString() };
    }
    if (obj.total !== null) {
      return { ...obj, total: (-parseFloat(obj.total)).toString() };
    }
    return { ...obj };
  }
```

- 5

```
  if (['+', '-', '*', '/', '%'].includes(buttonName)) {
    if (obj.operation !== null) {
      try {
        const total = operate(obj.total, obj.next, obj.operation);
        return {
          total: total.toString(),
          next: null,
          operation: buttonName,
        };
      } catch (error) {
        return {
          total: 'Error',
          next: null,
          operation: null,
        };
      }
    }
```

- lastly

```
if (buttonName === '/') {
    if (obj.next === '0') {
      return {
        total: 'Error',
        next: null,
        operation: null,
      };
    }
  }
```